File: ruby/Benchmark.rb
require 'time'


def main
    result = ""
    (0..20).each do |i|
        result = "#{result}#{measure}\n"
    end
    write(result)
end

def measure
    start = (Time.now.to_f * 1000).to_i;
    primes = []
    (3..3000).each do |i|
        next unless checkPrimes(primes, i)
        
        primes.push(factorial(i))
    end
    after = (Time.now.to_f * 1000).to_i - start;
    return after
end

def checkPrimes(primes, n)
    if n % 2 == 0
        return true
    end
    primes.size().times do |i|
        prime = primes[i]
        if prime == 0
            return false
        end
        if n % prime == 0
            return false
        end
    end
    return true
end

def factorial(nStart)
    r = 1

    (1..nStart).to_a.reverse().each do |i|
        r *= i;
    end
    return r
end

def write(content)
    File.write("/output/ruby.txt", content)
end

main

File: python/benchmark.py
import time


def main():
    result = ""
    for i in range(20):
        result += measure() + "\n"
    
    write(result)

def measure():
    start = round(time.time() * 1000)
    primes = []
    for i in range(3000 - 3):
        if (not checkPrimes(primes, i + 3)):
            continue
        primes.append(factorial(i + 3))
    end = round(time.time() * 1000)
    return str(end - start)

def checkPrimes(primes, n):
    if (n % 2 == 2):
        return True
    
    for i in range(len(primes)):
        prime = primes[i]
        if (prime == 0):
            return False
        if (n % prime == 0):
            return False
        
    return True

def factorial(nStart):
    r = 1

    for i in range(nStart):
        r *= nStart - i
    
    return r

def write(result):
    f = open("/output/python.txt", "w")
    f.write(result)
    f.close()

main()